src/server/stdio.ts

```diff
@@ -62,8 +62,19 @@ export class StdioServerTransport implements Transport {
   }
 
   async close(): Promise<void> {
+    // Remove our event listeners first
     this._stdin.off("data", this._ondata);
     this._stdin.off("error", this._onerror);
+
+    // Check if we were the only data listener
+    const remainingDataListeners = this._stdin.listenerCount('data');
+    if (remainingDataListeners === 0) {
+      // Only pause stdin if we were the only listener
+      // This prevents interfering with other parts of the application that might be using stdin
+      this._stdin.pause();
```

Comment:
I'm a bit surprised that this has any effect on termination, but then reading about [Node.js readable streams](https://nodejs.org/api/stream.html#readable-streams), I'm like 😵 

It feels a bit janky to do things this way, but I don't really have a better idea…

Reply:
Certainly not while messages are streaming in. The only question is whether there'd ever be a reason for a server implementation to want to read from stdin after terminating the normal MCP stuff. Probably not? But it's always hard to anticipate how people will use a library.